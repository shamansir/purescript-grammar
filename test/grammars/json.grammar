main :- value.
value :- (object | array | int | string | null).
int :- [[0-9], repSep([0-9], "")].
object :- ["{", repSep(keyValue, ","), "}"].
keyValue :- [string, ":", value].
array :- ["[", repSep(value, ","), "]"].
null :- "null".
string :- ["\"", repSep(stringChar, ""), "\""].
stringChar :- (^'"' | ['\\', '"']).
