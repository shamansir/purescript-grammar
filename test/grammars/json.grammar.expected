main :- value.
array :- ["[",repSep(value,","),"]"].
int :- [[0-9],repSep([0-9],"")].
keyValue :- [string,":",value].
null :- "null".
object :- ["{",repSep(keyValue,","),"}"].
string :- ["\"",repSep(stringChar,""),"\""].
stringChar :- (^'"'|['\\','"']).
value :- (object|array|int|string|null).